tools/dev/curl-converter.html

```diff
@@ -0,0 +1,259 @@
+      lines.push('print(response.json())');
+      return lines.join('\n');
+    }
+
+    function toGo(req) {
+      var lines = ['package main', '', 'import (', '  "fmt"', '  "net/http"', '  "io/ioutil"'];
+      if (req.data) lines.push('  "strings"');
+      lines.push(')', '', 'func main() {');
+      if (req.data) {
+        lines.push('  body := strings.NewReader(' + JSON.stringify(req.data) + ')');
+        lines.push('  req, _ := http.NewRequest(' + JSON.stringify(req.method) + ', ' + JSON.stringify(req.url) + ', body)');
+      } else {
+        lines.push('  req, _ := http.NewRequest(' + JSON.stringify(req.method) + ', ' + JSON.stringify(req.url) + ', nil)');
+      }
+      Object.keys(req.headers).forEach(function(k) {
+        lines.push('  req.Header.Set(' + JSON.stringify(k) + ', ' + JSON.stringify(req.headers[k]) + ')');
+      });
+      lines.push('', '  client := &http.Client{}');
+      lines.push('  resp, _ := client.Do(req)');
+      lines.push('  defer resp.Body.Close()');
+      lines.push('  data, _ := ioutil.ReadAll(resp.Body)');
+      lines.push('  fmt.Println(string(data))');
+      lines.push('}');
+      return lines.join('\n');
+    }
+
+    function toPHP(req) {
+      var lines = ['<?php', '', '$ch = curl_init();', '', 'curl_setopt($ch, CURLOPT_URL, ' + JSON.stringify(req.url) + ');'];
+      lines.push('curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);');
+      if (req.method !== 'GET') {
+        lines.push('curl_setopt($ch, CURLOPT_CUSTOMREQUEST, ' + JSON.stringify(req.method) + ');');
+      }
+      if (Object.keys(req.headers).length > 0) {
+        lines.push('curl_setopt($ch, CURLOPT_HTTPHEADER, [');
+        Object.keys(req.headers).forEach(function(k) {
+          lines.push('  ' + JSON.stringify(k + ': ' + req.headers[k]) + ',');
+        });
+        lines.push(']);');
+      }
+      if (req.data) {
+        lines.push('curl_setopt($ch, CURLOPT_POSTFIELDS, ' + JSON.stringify(req.data) + ');');
+      }
+      lines.push('', '$response = curl_exec($ch);', 'curl_close($ch);', '', 'echo $response;');
+      return lines.join('\n');
+    }
+
+    function toJava(req) {
+      var lines = ['import java.net.http.*;', 'import java.net.URI;', '', 'public class Main {', '  public static void main(String[] args) throws Exception {'];
+      lines.push('    HttpClient client = HttpClient.newHttpClient();');
+      lines.push('    HttpRequest.Builder builder = HttpRequest.newBuilder()');
+      lines.push('      .uri(URI.create(' + JSON.stringify(req.url) + '))');
+      Object.keys(req.headers).forEach(function(k) {
+        lines.push('      .header(' + JSON.stringify(k) + ', ' + JSON.stringify(req.headers[k]) + ')');
+      });
+      if (req.data) {
+        lines.push('      .method(' + JSON.stringify(req.method) + ', HttpRequest.BodyPublishers.ofString(' + JSON.stringify(req.data) + '));');
+      } else {
+        lines.push('      .' + req.method + '();');
```

Comment:
Invalid Java code for PUT/POST/PATCH without body. Java HttpRequest.Builder only has `.GET()` and `.DELETE()` as no-arg methods. For other methods, use `.method("PUT", HttpRequest.BodyPublishers.noBody())`.

```suggestion
      } else if (req.method === 'GET' || req.method === 'DELETE') {
        lines.push('      .' + req.method + '();');
      } else {
        lines.push('      .method(' + JSON.stringify(req.method) + ', HttpRequest.BodyPublishers.noBody());');
      }
```